orchestrator/core/llm/llm_agent_selector.py

```
@@ -848,51 +1051,352 @@ async def select_agents_for_subtasks(
+            required_skills = []
+        
+        required_tags = subtask.get('tags', [])
+        if isinstance(required_tags, str):
+            required_tags = [required_tags]
+        elif not required_tags:
+            required_tags = []
+        
+        requirements = [req for req in required_skills + required_tags if isinstance(req, str) and req.strip()]
+            
+        if not requirements:
+            logger.info(f"⚠️ Subtask {subtask_id} arrived with no required requirements (skills/tags). Will rely on agent_type={required_type}")
```

Ruff (0.14.4)

Undefined name `required_type`

(F821)